About us page

positiveimpact opened the pull request:
I implemented the about page which will simultaneously be used as a landing page during development.

I was unsure about the following things:
1. Where to store the "info" and "members" jsons, that are currently in AboutLayout.js
2. How to best go about deploying this during development. During development we only want this page  and a few select other pages to be accessible (onepager.pdf, faq).
3. On which level to import fontawesome. It will most likely be used on many pages, but maybe not on all. For now I only imported it on about.js as it's only being used there atm. But eventually all icons will be switched out by fontawesome icons, so maybe it would make sense to directly put it in layout.js.

And I haven't written test yet, but I decided to first get feedback on this and write test later, incase I need to change  a lot of things.

ReeceLangerock commented on src/components/general/Footer.js:
```diff
@@ -3,7 +3,22 @@ import styled from "styled-components";
   <Container>
+    <LeftSection>
+      <a href="imprint">Imprint</a>
+    </LeftSection>
```
What is imprint? And should it be a Link instead of an a tag?

positiveimpact replied:
Imprint contains some information about who's responsible for the website. It's legally required for any website in Germany. I will change it to a link and will add the actual page.

ReeceLangerock commented on src/layouts/AboutLayout.js:
```diff
@@ -0,0 +1,338 @@
+    <h3>{data.title}</h3>
+    <InfoText>{data.text}</InfoText>
+  </InfoBubbleContainer>
+);
```
Could probably make this it's own component, easier for testing and would cleanup this file some

ReeceLangerock replied:
Yeah would recommend that for sure especially with it having 4 styled-components

ReeceLangerock commented on src/layouts/AboutLayout.js:
```diff
@@ -0,0 +1,338 @@
+      {data.location}
+    </Location>
+  </MemberInfo>
+);
```
Same with this as own component

ReeceLangerock commented on src/layouts/AboutLayout.js:
```diff
@@ -0,0 +1,338 @@
+            <HeadlineText>CLIMATE CONNECT</HeadlineText>
+          </Headline>
+        </HeadlineContainer>
+      </AboutImageContainer>
```
Do we really need 4 custom divs here? Seems like it could probably just be 1 or 2.

positiveimpact replied:
I was able to remove one div, didn't find a way to remove more custom divs without losing vertical and horizontal centering as well as the box around the text.

ReeceLangerock commented:
Left a few comments, and to answer the questions you had:

1. I would create a data folder that can contain that as well as the projects json file. I would also organize the photos and icons. In the public folder do something like `public/images` and `public/icons`
2. We could simply comment out other pages for now. I don't think it would do any harm if someone dug around and found an unusable login page. Once we have authentication it'll be easy enough to just hide if the user isn't signed in but that's a future problem.
3. We probably want to use the [fontawesome react npm package](https://github.com/danawoodman/react-fontawesome) so we can just do `<FA name="rocket" />` instead of using the `<i>` tags. Then we just import the ones we need for that component. That way we don't import all of fontawesome either

ReeceLangerock commented on src/layouts/AboutLayout.js:
```diff
@@ -0,0 +1,338 @@
+              </a>
+            </Link>
+          </h2>
+        </LinkContainer>
```
These can be combined into a single render function, just with the data in a json or js object

ReeceLangerock commented on src/layouts/AboutLayout.js:
```diff
@@ -0,0 +1,338 @@
+  height: 150px;
+  margin: auto;
+  display: table;
+`;
```
Styling on the bubble and grid seems a bit weird, could probably be 3 or 4 in a row at fullscreen and the text should probably be split up some or made more concise and have a larger line-height. Just a big block of text for a the format it's in.

positiveimpact commented:
Thanks for your review, I implemented all the changes, that you suggested.

positiveimpact commented:
I will create another branch, comment out all other pages and deploy it on the server, that is hosting the climateconnect.earth.

positiveimpact commented:
This is a pretty old PR, that I finally got to rewrite to material UI. 
Back then I put a lot of different changes into one PR, I apologise for the mess and will create smaller PRs in the future. 

The PR includes the following things:
-added the About page as well as the About page link in the header
-added subfolders to the "public" folder to organise a bit more and adapted links everywhere
-added the impressum - this is a legally required page in Germany. I will create another PR for adding this and the social media icons to the footer.
-replaced the current logo being pulled from climateconnect.earth with a local file, that contains the new logo
-added new members to the team page
-added "layouts" and "themes" folder for the possibility of using different themes for different pages - at the moment the about page is the only page, that is using a custom theme and layout. This is mainly to be able to change some margins, e.g. to be able to stretch the image under the header over the whole page and to make sure the sizes of different headlines(h1, h2, ...) aren't affected on the about page, as the page pretty much stands on its own
-added tests for the about page

EvanHahn commented on src/components/about/AboutHeaderImage.js:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <Container className={classes.imageContainer} maxWidth={false}>
+      <Typography variant="h1" className={classes.headlineText}>
+        CLIMATE CONNECT
```
Can we make this say "Climate Connect" and then add `textTransform: "uppercase"` to the `headlineText` style above?

EvanHahn commented on src/components/about/AboutHeaderImage.js:
```diff
@@ -0,0 +1,41 @@
+  };
+});
+
+export default function AboutHeaderImage({ image }) {
```
I was a little surprised that we were passing this in as a prop instead of having the `AboutHeaderImage` "know" about which background image to use. Why is it passed in?

positiveimpact replied:
You're right, it doesn't really help. I changed it.

EvanHahn commented on src/components/about/AboutHeaderImage.js:
```diff
@@ -0,0 +1,41 @@
+      textAlign: "center"
+    }),
+    headlineText: {
+      color: "white",
```
This white text was hard to read on the image. Could we maybe add a drop shadow or something to make it easier to read? The `textShadow` attribute might help.

Alternatively, I'd be okay to remove this entire header because "Climate Connect" is already in the site's header.

positiveimpact replied:
I agree that in terms of text this header is not necessary at all, but a picture on the top of the page is visually appealing. I would keep it for now. I added textShadow to the text.

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+        <AboutHeaderImage image={{ url: backgroundImage }} />
+        <Typography variant="h4" color="primary" className={`${classes.root} ${classes.textBlock}`}>
+          We are an international team of volunteers building a non-profit climate action
+          webplatform
```
This should be "web platform", or could just be "platform".

positiveimpact replied:
changed it to "platform"

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+    <>
+      <Layout>
+        <AboutHeaderImage image={{ url: backgroundImage }} />
+        <Typography variant="h4" color="primary" className={`${classes.root} ${classes.textBlock}`}>
```
This can look like an `h4` but should be an `h1` for accessibility. Can we add the `component="h1"` prop to this?

```jsx
<Typography variant="h4" component="h1" ...
```

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+          webplatform
+        </Typography>
+        <Typography
+          variant="h3"
```
Like the above, this can look like an `h3` but should be an `h2` for accessibility. Can we add the `component="h2"` prop to this?

EvanHahn commented on public/data/info.json:
```diff
@@ -0,0 +1,27 @@
+[
```
I like this file, but I think the name is a bit wrong. It seems like it's something like `about_page_info.json` or similar.

EvanHahn commented on public/data/info.json:
```diff
@@ -0,0 +1,27 @@
+  {
+    "icon": "fa fa-chart-bar",
+    "title": "Measure the impact",
+    "text": "You are thinking about getting involved in climate action, but don't know how you can be most effective? On Climate Connect each action will have an impact score. This way you can find out how you can maximize your impact"
```
```diff
-You are thinking about getting involved in climate action, but don't know how you can be most effective? On Climate Connect each action will have an impact score. This way you can find out how you can maximize your impact
+You are thinking about getting involved in climate action, but don't know how you can be most effective? On Climate Connect, each action will have an impact score. This way you can find out how you can maximize your impact.

EvanHahn commented on public/data/info.json:
```diff
@@ -0,0 +1,27 @@
+  {
+    "icon": "fa fa-lightbulb",
+    "title": "Get inspired",
+    "text": "On Climate Connect, you will find blue prints of high-impact projects, that you will be able to replicate at your location. This way you can very easily find the most effective thing you can work on with your specific skillset."
```
```diff
-On Climate Connect, you will find blue prints of high-impact projects, that you will be able to replicate at your location. This way you can very easily find the most effective thing you can work on with your specific skillset.
+On Climate Connect, you will find blueprints of high-impact projects that you will be able to replicate at your location. That way, you can very easily find the most effective thing you can work on with your specific skills.
```

EvanHahn commented on public/data/info.json:
```diff
@@ -0,0 +1,27 @@
+  {
+    "icon": "fa fa-eye",
+    "title": "Maximum transparency",
+    "text": "Our code is available open source and can be found in our Github repository. The impact algorithm will be open source and the community will make the decisions on how to develop it further."
```
"Github" → "GitHub"

EvanHahn commented on public/data/links.json:
```diff
@@ -0,0 +1,20 @@
+[
+  {
+    "href": "documents/onepager.pdf",
```
Should this be an absolute link (`/documents/onepager.pdf` instead)?

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+const backgroundImage = `url("images/about_background.png")`;
+const useStyles = makeStyles(theme => {
+  return {
+    root: {
```
`root` implies that it's on the root element, but we're applying this in several places. Can we rename this to something like `centeredText`?

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+        </Container>
+        <Typography color="primary" variant="h6" className={classes.mainFocuses}>
+          Current main focuses:
+          <div>-Developing the platform (frontend: react, backend: node, postgres)</div>
```
1. This should be a `<ul>` instead of several `div`s, if possible.
1. React, Node, and Postgres should be capitalized.

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+          Current main focuses:
+          <div>-Developing the platform (frontend: react, backend: node, postgres)</div>
+          <div>-Finishing up the design</div>
+          <div>-Collecting user feedback: questionaires, interviews</div>
```
"questionaires" → "questionnaires"

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+          <div>-Developing the platform (frontend: react, backend: node, postgres)</div>
+          <div>-Finishing up the design</div>
+          <div>-Collecting user feedback: questionaires, interviews</div>
+          <div>-Developing our Social Media campaigns</div>
```
"Social Media" → "social media"

EvanHahn commented on pages/about.js:
```diff
@@ -0,0 +1,112 @@
+        </Typography>
+        <Typography color="primary" variant="h4" className={classes.root}>
+          If you would like to join our volunteer team, please send your application to
+          contact@climateconnect.earth
```
Could this be a link?
```jsx
<a href="mailto:contact@climateconnect.earth">
  contact@climateconnect.earth
</a>
```

EvanHahn commented on src/components/about/InfoLink.js:
```diff
@@ -0,0 +1,43 @@
+      <Typography variant="h4" color="primary" className={classes.root}>
+        <Link href={data.href}>
+          <a className={classes.linkText} target="_blank" rel="noopener noreferrer">
+            <Icon className={data.icon} />
```
Could we add a small amount of `marginRight` to these icons?

EvanHahn commented on src/components/about/InfoLink.js:
```diff
@@ -0,0 +1,43 @@
+    return (
+      <Typography variant="h4" color="primary" className={classes.root}>
+        <Link href={data.href}>
+          <a className={classes.linkText} target="_blank" rel="noopener noreferrer">
```
If it's an internal link, should it open in a new window like this?

positiveimpact replied:
You're right, it shouldn't. Changed it.

EvanHahn commented on src/components/about/InfoLink.js:
```diff
@@ -0,0 +1,43 @@
+  const classes = useStyles();
+
+  //use <Link> component only if link href is also on the climateconnect.earth domain
+  if (data.internal)
```
It looks like we might need to run the formatter here.

EvanHahn commented on test/components/about/aboutpage.test.js:
```diff
@@ -0,0 +1,35 @@
+describe("About page", () => {
+  const wrapper = mount(<AboutPage />);
+  it("contains correct infos", () => {
+    for (const i of info) {
```
If we rename `info` to something like `about_page_infos`, then we should consider renaming this variable from `i` to something more descriptive.

positiveimpact commented on package.json:
```diff
@@ -9,7 +9,9 @@
     "build": "next build",
-    "start": "next start"
+    "start": "next start",
+    "export": "next export",
```
I used this to be able to build the page statically for netlify. This is more part of deploying the pre-launch version (#73) though, so I'll remove it from this PR.

positiveimpact commented on yarn.lock:
```diff
@@ -27,12 +27,12 @@
   dependencies:
```
I might have accidentally run yarn upgrade. Reverted the change.

positiveimpact commented on src/components/layouts/aboutLayout.js:
```diff
@@ -0,0 +1,41 @@
+      <Head>
+        <link
+          rel="stylesheet"
+          href="https://use.fontawesome.com/releases/v5.11.2/css/all.css"
```
I considered it. The main reason why I kept the fontawesome icons was because I didn't know how to dynamically import the icons that the info.json and links.json files tell me to import.

EvanHahn replied:
Personally, I think it'd be best to use only one set of icons. That will make things less confusing for us, and help us standardize on one thing visually.

We could solve this in one of two ways:

1. Convert `info.json` to use JavaScript (`info.js` or similar). Then you could do something like this:

   ```jsx
   // ...
   {
      title: "Share your actions",
      icon: PlusIcon,
      // ...
   ```

2. Add something that maps an icon to its Material UI icon. This would take a string and convert it into the Material UI component. Something like:

   ```js
   const materialIconByName = {
      plus: PlusIcon,
      chart: ChartIcon,
      // ...
   };
   ```

positiveimpact replied:
I will look to replace the FA icons with material ui native icons.

positiveimpact replied:
I replaced them with Material UI icons using option 1

positiveimpact commented on pages/create.js:
```diff
@@ -2,9 +2,9 @@ import React from "react";
-  <Layout title="Create a Project">
-    <h1>Create a Project</h1>
-  </Layout>
+  <div>
```
removed it.

positiveimpact commented on src/themes/aboutTheme.js:
```diff
@@ -0,0 +1,27 @@
+import { createMuiTheme } from "@material-ui/core/styles";
+
+const theme = createMuiTheme({
```
Doesn't seem to work on first try. I'll look further into this later tonight.

positiveimpact commented:
Thanks for the review! I implemented all the changes you suggested, but left in the header image. I will address the 2 unresolved issues later tonight. 
I replaced the pngs with jpgs, would be nice if you could optimize them for size.

positiveimpact commented:
Converted the two transparent logos back to pngs. 
Implemented all changes.

positiveimpact commented:
After replacing Fontawesome Icons with material UI icons, I had some problems testing for their existence. I solved that with a workaround by adding a name property to each icon and testing for that property. 

This PR is ready for review again.

ddhanesha commented on pages/about.js:
```diff
@@ -0,0 +1,119 @@
+      margin: "0 auto"
+    }
+  };
+});
```
Not entirely sure about the best practice so correct me if I am wrong. But should we put the styles const into a separate file? Maybe we can have styles directory and then import files from that directory. cc/ @EvanHahn

positiveimpact replied:
I feel like it's easier to work with this way

ddhanesha replied:
Yeah, I am fine with it. Just wanted to confirm.

ddhanesha commented on pages/faq.js:
```diff
@@ -0,0 +1,10 @@
+  <div>
+    <Layout title="FAQ" />
+  </div>
+);
```
We could also add a small message here that FAQs are coming soon?

positiveimpact replied:
Added a Coming Soon message

positiveimpact commented:
Implemented the suggested changes. This should be ready for merge if there are no other issues.

EvanHahn commented:
Planning to take a look today. I'll merge if it looks good.

EvanHahn commented on public/data/projects.json:
```diff
@@ -3,9 +3,9 @@
       "id": "78b6677c-28d2-4b2b-9825-4479706d2402",
       "name": "CO2-Labels for University Canteen",
-      "image": "/images/canteen.png",
+      "image": "images/canteen.jpg",
```
Should these have changed?

positiveimpact replied:
I put back in the "/"

EvanHahn commented on src/components/layouts/AboutLayout.js:
```diff
@@ -0,0 +1,42 @@
+  return (
+    <>
+      <Head>
+        <link
```
We're no longer using Font Awesome (I think), so we can remove this.

positiveimpact replied:
removed
